### ISSUE TITLE
ResNet on MNIST

### ISSUE THREAD

**Issue by Akaame** [FluxML/Metalhead.jl#47]
How would one apply ResNet on Kaggle's MNIST dataset? I shaved off the final dense layer and the softmax layer ( ResNet()[:20] ). I get a dataset of (42000, 2048).   From there on train the following net.

Dense(2048, 1024, relu),
BatchNorm(1024),
Dense(1024, 512, relu),
BatchNorm(512),
Dense(512, 256, relu),
BatchNorm(256),
Dense(256, 10), 
softmax

This does not surpass %88 training accuracy even after training for 50 epochs whereas a regular convolutional network I train from scratch for a few epoch easily gets %99.5 train accuracy. More generally, how would you apply ResNet(or any other) net to a custom dataset? Can I get a minimal example.

Thanks in advance.

**Comment by SebastianCallh** [comment]
Did you figure this out? I am trying to use CIFAR10 and feed it to the provided ResNet but I'm not sure how to deal with the type Array{ColorTypes.RGB...} that the images have. I managed to fed a tensor
`x = rand(Float32, 224, 224, 3, 1)` to the model, so the question is how can I stack the different RGB channels from the `ColorTypes.RGB`?

**Comment by avik-pal** [comment]
@SebastianCallh You can use the [channelview](https://juliaimages.org/latest/function_reference/#ImageCore.channelview) function.

**Comment by avik-pal** [comment]
@Akaame I am not surprised at such a low accuracy. Resnet is pre-trained on Imagenet, so the 2048 dimension feature representation that you obtain for the MNIST dataset, would not be "good enough" to accurately classify them.

**Comment by SebastianCallh** [comment]
@avik-pal  thank you for your response.

`channelview` gives me a CxWxH tensor, and I now need to get it into BxHxWxC shape for Flux. I have found the functions `permuteddimsview` and `permutedims` but none of them seem to solve the problem. Could you please show me how you would do to feed an image to Flux?

**Comment by avik-pal** [comment]
https://github.com/FluxML/Metalhead.jl/blob/master/src/utils.jl#L63-L65 does a similar transform
